src/Nethermind/Nethermind.TxPool/Filters/TooExpensiveTxFilter.cs

```diff
@@ -59,10 +50,7 @@ public AcceptTxResult Accept(Transaction tx, TxFilteringState state, TxHandlingO
                 }
             }
 
-            UInt256 affordableGasPrice = tx.CalculateAffordableGasPrice(spec.IsEip1559Enabled, _headInfo.CurrentBaseFee, balance > cumulativeCost ? balance - cumulativeCost : 0);
```

Comment:
Isn't the balance check needed or you could jump the queue with the high gaslimit or priority fee that you couldn't pay for?